### PR TITLE
Sprint 9

### PR DESCRIPTION
**Pull request by RastelliBru**

### CODE REVIEW

**Review by wandersonmaceds — APPROVED** [review]

File: account/src/helpers/password.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import bcrypt from 'bcryptjs';
+
+export async function hashPassword(password) {
+    const hashedPassword = await bcrypt.hash(password, 10);
```
Comment (DPechetti) [inline account/src/helpers/password.js]: I think it's a better approach to put the salt in the .env file or get it from the caller

File: account/src/use-case/createUserAccount.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
 import crypto from 'crypto'
 import { saveAccount } from "../../repositories/accountRepository.js";
+import bcrypt from 'bcryptjs'
 
 export async function createUserUseCase(name, email, password) {
   const createdDate = new Date().toISOString().substring(0, 10);
+  const hashedPassword = bcrypt.hashSync(password, 10);
```
Comment (DPechetti) [inline account/src/use-case/createUserAccount.js]: I think it's a better approach to put the salt in the .env file or get it from the caller

File: account/src/use-case/createUserToken.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    if(!possibleUser){
+        return null;
+    }
+
+    const passwordCheck = await comparePassword(password, possibleUser.password);
+
+    if(passwordCheck) {
+        return generateToken(possibleUser.id);
+    }
```
Comment (DPechetti) [inline account/src/use-case/createUserToken.js]: i suggest you always put the happy flow at the end

**Review by DPechetti — COMMENTED** [review]
just suggestions

**Review by DPechetti — APPROVED** [review]

File: account/src/helpers/password.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import bcrypt from 'bcryptjs';
+
+export async function hashPassword(password) {
+    const hashedPassword = await bcrypt.hash(password, parseInt(process.env.HASH_SALT));
```
Comment (DPechetti) [inline account/src/helpers/password.js:4]: i think the right approach is to create a variable with the result before and put the variable here after

File: account/src/use-case/createUserAccount.js [hunk 5]
```diff
@@ -1,14 +1,17 @@
 import crypto from 'crypto'
 import { saveAccount } from "../../repositories/accountRepository.js";
+import bcrypt from 'bcryptjs'
+
 
 export async function createUserUseCase(name, email, password) {
   const createdDate = new Date().toISOString().substring(0, 10);
+  const hashedPassword = bcrypt.hashSync(password, parseInt(process.env.HASH_SALT));
```
Comment (DPechetti) [inline account/src/use-case/createUserAccount.js:8]: i think the right approach is to create a variable with the result before and put the variable here after

File: docker-compose.yml [hunk 6]
```diff
@@ -5,6 +5,7 @@ services:
       - mysqldb_products
     image: pagonxt/product
     build: product
+    entrypoint: 'npm run start:dev'
```
Comment (DPechetti) [inline docker-compose.yml:8]: your docker cannot come up with dev script

File: docker-compose.yml [hunk 7]
```diff
@@ -15,6 +16,7 @@ services:
       - mongodb_accounts
     image: pagonxt/account
     build: account
+    entrypoint: 'npm run start:dev'
```
Comment (DPechetti) [inline docker-compose.yml:19]: your docker cannot come up with dev script

**Review by DPechetti — COMMENTED** [review]
some adjusts to improve your code => sprint 9

**Pull request merged**